In [6]:
## import packages
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
## read dataset (.data) as CSV 
## put into dataframe
## add headers
## make a copy

heart_dataframe = pd.read_csv('processed.cleveland.data', header=None)

heart_data_1 = heart_dataframe.copy()
heart_data_1.columns = ['age','sex','cp','trestbps','chol','fbs','restecg','thalach',
                      'exang','oldpeak','slope','ca','thal','num']
#heart_data_1.to_csv ('processed.cleveland.data', index=None)

In [4]:
heart_data_1.head()

age  sex   cp  trestbps   chol  fbs  restecg  thalach  exang  oldpeak  \
0  63.0  1.0  1.0     145.0  233.0  1.0      2.0    150.0    0.0      2.3   
1  67.0  1.0  4.0     160.0  286.0  0.0      2.0    108.0    1.0      1.5   
2  67.0  1.0  4.0     120.0  229.0  0.0      2.0    129.0    1.0      2.6   
3  37.0  1.0  3.0     130.0  250.0  0.0      0.0    187.0    0.0      3.5   
4  41.0  0.0  2.0     130.0  204.0  0.0      2.0    172.0    0.0      1.4   

   slope   ca thal  num  
0    3.0  0.0  6.0    0  
1    2.0  3.0  3.0    2  
2    2.0  2.0  7.0    1  
3    3.0  0.0  3.0    0  
4    1.0  0.0  3.0    0

In [5]:
print('Heart Disease data set dimensions: {}'.format(heart_data_1.shape))

Heart Disease data set dimensions: (303, 14)


14 Attributes used

#3  (age)      
#4  (sex): 1 = male, 0 = female     
#9  (cp): cp: chest pain type
            Value 1: typical angina
            Value 2: atypical angina
            Value 3: non-anginal pain
            Value 4: asymptomatic        
#10 (trestbps): resting blood pressure (in mm Hg on admission to the hospital)  
#12 (chol): serum cholestoral in mg/dl  
#16 (fbs): (fasting blood sugar > 120 mg/dl)  (1 = true; 0 = false)       
#19 (restecg): restecg: resting electrocardiographic results
                Value 0: normal
                Value 1: having ST-T wave abnormality 
                          (T wave inversions and/or ST elevation or depression of > 0.05 mV)
                Value 2: showing probable or definite left ventricular hypertrophy by Estes' criteria)   
#32 (thalach): 32 thalach: maximum heart rate achieved   
#38 (exang): exercise induced angina (1 = yes; 0 = no)     
#40 (oldpeak): ST depression induced by exercise relative to rest   
#41 (slope): the slope of the peak exercise ST segment
            Value 1: upsloping
            Value 2: flat
            Value 3: downsloping      
#44 (ca): number of major vessels (0-3) colored by flourosopy        
#51 (thal): 3 = normal; 6 = fixed defect; 7 = reversable defect       
#58 (num): diagnosis of heart disease (angiographic disease status)  (the predicted attribute)
            Value 0: < 50% diameter narrowing
            Value 1: > 50% diameter narrowing
        (in any major vessel: attributes 59 through 68 are vessels)
        
      
    
    